### ISSUE TITLE
Don't suggest unchanged files for `git add <tab>`

### ISSUE THREAD

**Issue by johanhelsing** [dahlbyk/posh-git#777]
### System Details

- posh-git version/path: 1.0.0 beta4 ~\OneDrive - Company AS\Documents\PowerShell\Modules\posh-git\1.0.0
- PowerShell version: 7.0.0
- git version 2.25.1.windows.1
- OS: Microsoft Windows NT 10.0.18363.0

### Issue Description

When typing `git add <tab>` I get suggestions for files that are neither modified nor untracked.

Other git auto-completion implementations, such as fish's will only suggest paths that are meaningful to stage, such that if you have a file at some deep nesting level, you can complete it with a single tab. i.e.

```
git add some/file/ridiculously/deeply/nested/with/lots/of/similar/siblings.txt
```

**Comment by dahlbyk** [comment]
Assuming this is not still an issue. As @zanseb said, we can't complete interesting file names without `EnableFileStatus` enabled.

**Comment by johanhelsing** [comment]
Still doesn't work for me:

```
dev/poshgittest git init
Initialized empty Git repository in C:/Users/Johan/dev/poshgittest/.git/
dev/poshgittest main wsl mkdir -p a/b
dev/poshgittest main wsl touch a/b/c.txt
dev/poshgittest main …1 wsl touch a/d.txt
dev/poshgittest main …1 git add .
dev/poshgittest main →2 git commit -m "Initial commit"
[main (root-commit) 0c6ef3d] Initial commit
 2 files changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 a/b/c.txt
 create mode 100644 a/d.txt
dev/poshgittest main echo 123 > a/b/c.txt
dev/poshgittest main ~1 git add .\a\
b      d.txt
dev/poshgittest main ~1 git add .\a\^C
dev/poshgittest main ~1 git st
On branch main
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   a/b/c.txt

no changes added to commit (use "git add" and/or "git commit -a")
dev/poshgittest main ~1 cat $profile | ag GitPrompt
$GitPromptSettings.EnableFileStatus = $True
dev/poshgittest main ~1 wsl fish
DESKTOP-GA24L3J
/m/c/U/J/d/poshgittest (main ✎1) git add a/b/c.txt
```

First, I use powershell and posh-git. It still suggests `a/d.txt` even though it's unchanged.

The last line is fish's git auto-completion (I hit tab after typing `git add a` and it completed to the only changed file within the repo and correctly ignored `a/d.txt`.

**Comment by dahlbyk** [comment]
What does `$GitStatus` show at that point?

It looks like your prompt is customized without posh-git, which means you're probably missing the behind-the-scenes `status` call that populates the path list for tab expansion.

**Comment by johanhelsing** [comment]
Yeah, I'm using a cross-shell rusty thingy for status and only using posh-git only for completion. Is there a way to trigger that call?

**Comment by dahlbyk** [comment]
What does your PowerShell `prompt` function look like?

We could conceivably look up `$GitStatus` on demand if it's missing when tab expansion requires it.

**Comment by dahlbyk** [comment]
Try adding `$global:GitStatus = Get-GitStatus` before your `Prompt` returns.

**Comment by johanhelsing** [comment]
Yep, that works! Thanks :)

**Comment by dahlbyk** [comment]
Great!

We can leave this open should someone feel compelled to teach tab expansion to `Get-GitStatus` on demand in tab expansion, if it's not already available from the posh-git `prompt`.
